odoo/addons/base/tests/test_views.py

```diff
@@ -674,6 +674,51 @@ def test_branding_inherit(self):
             second.get('data-oe-id'),
             "second should come from the extension view")
 
+    def test_branding_primary_inherit(self):
+        view1 = self.View.create({
+            'name': "Base view",
+            'type': 'qweb',
+            'arch': """<root>
+                <item order="1"/>
+            </root>
+            """
+        })
+        view2 = self.View.create({
+            'name': "Extension",
+            'type': 'qweb',
+            'mode': 'primary',
+            'inherit_id': view1.id,
+            'arch': """<xpath expr="//item" position="before">
+                <item order="2"/>
+            </xpath>
+            """
+        })
+
+        arch_string = view2.with_context(inherit_branding=True).read_combined(['arch'])['arch']
+
+        arch = etree.fromstring(arch_string)
+        self.View.distribute_branding(arch)
+
+        [initial] = arch.xpath('//item[@order=1]')
+        self.assertEqual(
+            str(view1.id),
```

Comment:
While it works, you invert the operands. Your know value should be the second one.

Reply:
ok, I did the same as other test in the file. I will adapt